src/Tempest/Http/HttpApplication.php

```diff
@@ -16,33 +17,29 @@
 
 final readonly class HttpApplication implements Application
 {
-    public function __construct(
-        private Container $container,
-        private AppConfig $appConfig,
-    ) {
-    }
+    public function __construct(private Container $container) {}
 
-    public static function boot(string $root, ?AppConfig $appConfig = null): self
+    /** @param \Tempest\Core\DiscoveryLocation[] $discoveryLocations */
+    public static function boot(
+        string $root,
+        array $discoveryLocations = [],
+    ): self
     {
-        $container = Tempest::boot($root, $appConfig);
-
-        // TODO: check if we need this
-        $appConfig = $container->get(AppConfig::class);
+        $container = Tempest::boot($root, $discoveryLocations);
 
         // Application,
         // TODO: can be refactored to resolve via the container
-        $application = new HttpApplication(
-            container: $container,
-            appConfig: $appConfig,
-        );
+        $application = new HttpApplication($container);
 
         $container->singleton(Application::class, fn () => $application);
 
+        $root = $container->get(Kernel::class)->root;
+
         // Application-specific setup
         $logConfig = $container->get(LogConfig::class);
-        $logConfig->debugLogPath = PathHelper::make($appConfig->root, '/log/debug.log');
+        $logConfig->debugLogPath = PathHelper::make($root, '/log/debug.log');
         $logConfig->serverLogPath = env('SERVER_LOG');
-        $logConfig->channels[] = new AppendLogChannel(PathHelper::make($appConfig->root, '/log/tempest.log'));
+        $logConfig->channels[] = new AppendLogChannel(PathHelper::make($root, '/log/tempest.log'));
 
         return $application;
```

Comment:
This feels like setup that could be in the kernel?

Reply:
I'm not sure… wouldn't logging be application specific? OTOH, I'm fine moving it to kernel first and see whether it actually becomes necessary to complicate further

Reply:
Come to think of it... do we really need to do any setup here or can that be in a config?

Reply:
That's a good point, though it's outside the scope of this PR I think